test/clojure/core_test/persistent_bang.cljc

```diff
@@ -0,0 +1,34 @@
+(ns clojure.core-test.persistent-bang
+  (:require [clojure.test :refer [are deftest testing]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists]]))
+
+(when-var-exists persistent!
+  (deftest test-persistent!
+
+    (testing "map"
+      (are [expected coll] (= expected (persistent! coll))
+                           {} (transient {})
+                           {nil nil} (transient {nil nil})
+                           {:a 1 :b 2} (transient {:a 1 :b 2})))
+
+    (testing "vector"
+      (are [expected coll] (= expected (persistent! coll))
+                           [] (transient [])
+                           [nil] (transient [nil])
+                           [1 2 3] (transient [1 2 3])))
+
+    (testing "set"
+      (are [expected coll] (= expected (persistent! coll))
+                           #{} (transient #{})
+                           #{nil} (transient #{nil})
+                           #{:a :b :c} (transient #{:a :b :c})))
+
```

Comment:
I think we'll want a test for calling `persistent!` multiple times on a transient, since that will throw. We don't need to do it for every possible shape, but we do want to capture that behavior, since it's documented.